iFixit/checkout-ui-extensions
# Purchase orders: Support it

Closes https://github.com/iFixit/checkout-ui-extensions/issues/1

Summary
--
- Admin order page now sees purchase order attribute
- Checkout now has purchase order input field as a dynamic field
- Purchase order field is visible on thank you page and order status page

CR
--
:question: I had a few concerns that came up during development:
- Currently, if a user selects the Purchase Order option at checkout, fills in the Purchase Order field, then switches to a different payment option, the order will have the purchase order attribute with a value of `''`. It seems like there isn't any way to remove an attribute once it's been applied. Is this fine?
- To check if the payment method is Purchase Order, I am checking if the payment type is of `manualPayment`. Is it possible in the future that we will have other custom payment methods that have the type `manualPayment`?

QA
--
Note that [there may be a bug ](https://github.com/Shopify/ui-extensions/issues/1290)with the purchase order disappearing from the Thank You page if you reload it
Similar testing to https://github.com/iFixit/checkout-ui-extensions/pull/3
- Include the url param `?placement-reference=PAYMENT4`
- In Admin order view, make sure you can see the purchase order attribute
- Note that Shopify [does not currently allow us](https://help.shopify.com/en/partners/dashboard/managing-stores/test-orders-in-dev-stores) to place test orders with manual payment methods. We have to test this in a "hacky" way, as described below.

When placing an order:
1. Comment out `extensions/ui-extension/src/fields/PurchaseOrder.tsx:43`.
2. When selecting your payment method, select Purchase Order. Then observe the input field pops up at the bottom.
3. Fill in the input with any value
4. Then, select credit card and fill in the bogus credit card values (credit card number: 1, cvv: any, name: any, exp: any)
5. Checkout, and observe at the Thank You page (page after you complete checkout), there is a Purchase Order field displayed with the value you filled earlier.
